steps/s3/session.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2021 Telefonica Cybersecurity & Cloud Tech SL
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3steps
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Telefonica/golium"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	aws_s "github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/sirupsen/logrus"
+)
+
+type Session struct {
+	Client           *aws_s.Session
+	CreatedDocuments []*CreatedDocument
+}
+
+type CreatedDocument struct {
+	bucket string
+	key    string
+}
+
+// newS3Session initiates a new aws session.
+func (s *Session) newS3Session(ctx context.Context) error {
+	logger := GetLogger()
+	logger.LogMessage("Creating a new S3 session")
+	var err error
```

Comment:
Better to move this line where it's really used (L63)

Reply:
Done

Reply:
I cannot see it, am i wrong?

Reply:
Sorry, I was refering to `var err error`.
To move it to L81, where is close to where you use it: `s.Client, err = ... `

Reply:
OK! Done